WiserTaskScheduler/WiserTaskScheduler/Core/Workers/BaseWorker.cs

```diff
@@ -138,23 +143,35 @@ protected override async Task ExecuteAsync(CancellationToken stoppingToken)
 
                             if (!String.IsNullOrWhiteSpace(ConfigurationName))
                             {
-                                var states = await wiserDashboardService.GetLogStatesFromLastRun(ConfigurationName, RunScheme.TimeId, runStartTime);
-
-                                var state = "success";
-                                if (await wiserDashboardService.IsServicePaused(ConfigurationName, RunScheme.TimeId))
+                                paused = await wiserDashboardService.IsServicePaused(ConfigurationName, RunScheme.TimeId);
+                                // only set state to paused if we are sure it's paused
+                                if (paused.HasValue && paused.Value)
                                 {
                                     state = "paused";
                                 }
-                                else if (states.Contains("Critical", StringComparer.OrdinalIgnoreCase) || states.Contains("Error", StringComparer.OrdinalIgnoreCase))
+                                else
                                 {
-                                    state = "failed";
+                                    var states = await wiserDashboardService.GetLogStatesFromLastRun(ConfigurationName, RunScheme.TimeId, runStartTime);
+                                    if (states.Contains("Critical", StringComparer.OrdinalIgnoreCase) || states.Contains("Error", StringComparer.OrdinalIgnoreCase))
+                                    {
+                                        state = "failed";
+                                    }
+                                    else if (states.Contains("Warning", StringComparer.OrdinalIgnoreCase))
+                                    {
+                                        state = "warning";
+                                    }
+                                    else if (states.Count == 0) // exception occurred in getting states
```

Comment:
We discussed to use an empty list for this. But during the review I remembered a situation that can cause this state to be triggered accidently. If the log settings have been set to only log warnings or higher the list would be empty if everything went correctly. Maybe it is better to give a "null" list back or with a single entry with state "Unknown" when it failed to execute the query.

Reply:
Used the "null" list solution to keep it consistent with the other calls that can fail.